crates/ruff_python_trivia/src/tokenizer.rs

```diff
@@ -762,322 +856,133 @@ impl<'a> SimpleTokenizer<'a> {
-                        '=' | '*' | '/' | '@' | '!' | '<' | '>' | '.' => {
-                            // This could be a single-token token, like `+` in `x + y`, or a
-                            // multi-character token, like `+=` in `x += y`. It could also be a sequence
-                            // of multi-character tokens, like `x ==== y`, which is invalid, _but_ it's
-                            // important that we produce the same token stream when lexing backwards as
-                            // we do when lexing forwards. So, identify the range of the sequence, lex
-                            // forwards, and return the last token.
-                            let mut cursor = self.cursor.clone();
-                            cursor.eat_back_while(|c| {
-                                matches!(
-                                    c,
-                                    ':' | '~'
-                                        | '%'
-                                        | '|'
-                                        | '&'
-                                        | '^'
-                                        | '+'
-                                        | '-'
-                                        | '='
-                                        | '*'
-                                        | '/'
-                                        | '@'
-                                        | '!'
-                                        | '<'
-                                        | '>'
-                                        | '.'
-                                )
-                            });
-
-                            let token_len = cursor.token_len();
-                            let range = TextRange::at(self.back_offset - token_len, token_len);
-
-                            let forward_lexer = Self::new(self.source, range);
-                            if let Some(token) = forward_lexer.last() {
-                                // If the token spans multiple characters, bump the cursor. Note,
-                                // though, that we already bumped the cursor to past the last character
-                                // in the token at the very start of `next_token_back`.
-                                for _ in self.source[token.range].chars().rev().skip(1) {
-                                    self.cursor.bump_back().unwrap();
-                                }
-                                token.kind()
-                            } else {
-                                self.bogus = true;
-                                SimpleTokenKind::Other
-                            }
-                        }
-
-                        _ => {
-                            self.bogus = true;
-                            SimpleTokenKind::Other
-                        }
-                    }
-                }
-            }
+            '#' => unreachable!(),
```

Comment:
This could potentially happen if we start lexing before the `expr` in 

```python
f"#{expr}"
```

Let's return `Other` in this case.

Reply:
ok, but it's really bad if you're doing that, the output of the tokenizer is completely wrong when you're lexing string content as rust